torch/nn/modules/rnn.py

```diff
@@ -252,11 +252,20 @@ def flatten_parameters(self) -> None:
         # a sufficient check, because overlapping parameter buffers that don't completely
         # alias would break the assumptions of the uniqueness check in
         # Module.named_parameters().
-        unique_data_ptrs = {
-            p.data_ptr() for p in self._flat_weights  # type: ignore[union-attr]
-        }
-        if len(unique_data_ptrs) != len(self._flat_weights):
-            return
+        try:
+            from torch.multiprocessing.reductions import StorageWeakRef
```

Comment:
StorageWeakRef implementation actually doesn't look like it is multiprocessing specific, so you should move it out of there. It is also not good practice to do local imports like this. @bdhirsh What do you think about defining in torch/__init__.py?

Reply:
Here would be the right place https://github.com/pytorch/pytorch/blob/43a09189c68fe02bd9d8433c4a144ffc9bbf895c/torch/utils/weak.py#L322